lib/outline.vim

```diff
@@ -231,6 +234,7 @@ def UpdateOutline()
 enddef
 
 
+# TODO: probably Quite Slow if >8000 lines! required to refine!
```

Comment:
If I remember peoperly, I went the brute force method there because in vim you can edit any line of any buffer from any position, and that gives you lot of freedom but it is also a curse for optimizing because you have no information. What if the cursor  is at line 3 and someone (user or some plugin) write e.g. from lines 100 to 150 with e.g. setline() or, even worst with setbufline()? 
This was just an example, but in Vim too many things can happen, and for that reason I preferred the conservative, bulletproof way. 
If you have ideas on how to keep it bulletproof and at the same time making it faster you are welcome to suggest!